tests/unit/src/test/scala/tests/DocumentHighlightSlowSuite.scala

```diff
@@ -0,0 +1,141 @@
+package tests
+
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+
+object DocumentHighlightSlowSuite extends BaseSlowSuite("documentHighlight") {
+
+  fileNamedTestAsync("singleHighlight") { fileName =>
+    for {
+      _ <- server.initialize(
+        s"""/metals.json
+           |{"a":{}}
+           |/$fileName
+           |object Main {
+           |  Option(1).head
+           |}
+        """.stripMargin
+      )
+      _ <- server.didOpen(fileName)
+      _ <- server.assertHighlight(
+        fileName,
+        """
+          |object Main {
+          |  Option(1).he@@ad
+          |}""".stripMargin,
+        """
+          |object Main {
+          |  Option(1).<<head>>
+          |}""".stripMargin
+      )
+    } yield ()
+  }
+
+  fileNamedTestAsync("multipleHighlight") { fileName =>
+    for {
+      _ <- server.initialize(
+        s"""/metals.json
+           |{"a":{}}
+           |/$fileName
+           |object Main {
+           |  val abc = 123
+           |  abc.toInt
+           |  println(abc)
+           |}
+        """.stripMargin
+      )
+      _ <- server.didOpen(fileName)
+      _ <- server.assertHighlight(
+        fileName,
+        """
+          |object Main {
+          |  val abc = 123
+          |  abc.toInt
+          |  println(ab@@c)
+          |}""".stripMargin,
+        """
+          |object Main {
+          |  val <<abc>> = 123
+          |  <<abc>>.toInt
+          |  println(<<abc>>)
+          |}""".stripMargin
```

Comment:
Cool tests! 😍 
It would be interesting to also have something along the lines of

```
object Main {
  val <<a>> = 123
  val f = (a: Int) => abc + 1
  println(<<a>>)
}
```

Reply:
Added the test case - also simplified it a bit, so should be easy to spot